### PR TITLE
Fix build errors

### PR DESCRIPTION
**Pull request by lisajian**
As the name implies, fixes remaining build errors from rebasing against mainline

This change is pending fixes against mainline build errors (`Error: ENOENT: no such file or directory ...`)

### CODE REVIEW

File: src/components/Auth/UserFormDialog/controls/Email.tsx [hunk 1]
```diff
@@ -77,6 +77,9 @@ export const Email: React.FC<
     pattern: EMAIL_REGEX,
   });
 
+  // TODO: Only adding this todo to get reviewer visibility. I added this
```
Comment (lisajian) [inline src/components/Auth/UserFormDialog/controls/Email.tsx:80]: @tjlav5

Comment (tjlav5) [reply src/components/Auth/UserFormDialog/controls/Email.tsx:80]: Curious, why does it feel wonky?

File: src/components/Auth/UserFormDialog/controls/MultiFactorAuth.tsx [hunk 2]
```diff
@@ -116,7 +123,7 @@ export const MultiFactor: React.FC<
           </Typography>
           <div>
             {fields.map((item, index) => {
-              const fieldName = `mfaPhoneInfo[${index}].phoneInfo`;
+              const fieldName = `mfaPhoneInfo.${index}.phoneInfo` as const;
```
Comment (tjlav5) [inline src/components/Auth/UserFormDialog/controls/MultiFactorAuth.tsx:126]: No need for `as const` on a string?

File: src/components/Auth/UserFormDialog/controls/MultiFactorAuth.tsx [hunk 3]
```diff
@@ -126,11 +133,17 @@ export const MultiFactor: React.FC<
                   } else if ((error as any).phoneInfo.type === 'required') {
                     return 'Phone number is required';
                   } else {
-                    return error.message;
+                    return (error as any).phoneInfo.message;
                   }
                 }
               };
 
+              // TODO: ask yuchen
```
Comment (tjlav5) [inline src/components/Auth/UserFormDialog/controls/MultiFactorAuth.tsx:141]: nit: Could you elaborate or remove the TODO?

**Review by tjlav5 — DISMISSED** [review]

File: src/components/Auth/UserFormDialog/controls/MultiFactorAuth.tsx (hunk 2 above)
Comment (yuchenshi) [reply src/components/Auth/UserFormDialog/controls/MultiFactorAuth.tsx]: FWIW, `as const` narrows the literal type to even more specific than `string`. The actual type is something more like `mfaPhoneInfo.${number}.phoneInfo` (i.e. TypeScript knows that it is a string with the format of a number between two fixed parts), which is needed for the library.

**Review by yuchenshi — APPROVED** [review]

File: src/components/Auth/UserFormDialog/controls/Email.tsx (hunk 1 above)
Comment (yuchenshi) [reply src/components/Auth/UserFormDialog/controls/Email.tsx]: Now fixed

File: src/store/auth/selectors.ts [hunk 4]
```diff
@@ -53,9 +53,21 @@ export const getUsers = createSelector(getAuthUsers, (users) => {
...
+export const getAllTakenEmails = createSelector(
+  getCurrentEditedUser,
+  getUsers,
+  (currUser: AuthUser | undefined, users: AuthUser[]) => {
+    if (!currUser) {
+      return new Set(users.map((u) => u.email).filter((e) => !!e));
+    }
+    return new Set(
+      users
+        .filter((user) => user.localId !== currUser.localId)
```
[the remark this answers is not in the document]
Comment (lisajian) [reply src/store/auth/selectors.ts]: Done!

**Pull request merged**